src/header/Header.js

```diff
@@ -1,123 +1,88 @@
+    ) : children.title ? (
+      <Text {...children}>{children.title}</Text>
+    ) : children.icon ? (
+      <Icon
+        {...children}
+        containerStyle={[
+          { alignItems: ALIGN_STYLE[placement] },
+          children.containerStyle,
         ]}
-      >
-        <Title {...value} />
-      </View>
-    ) : (
-      <NavButton {...value} key={type} />
-    );
-  }
-  return type === 'center' ? null : <DummyNavButton key={type} />;
-}
-
-function populateChildren(propChildren, placement, centerComponentStyle) {
-  const childrenArray = [];
-
-  const leftComponent = generateChild(propChildren.leftComponent, 'left');
-  const centerComponent = generateChild(
-    propChildren.centerComponent,
-    'center',
-    placement,
-    centerComponentStyle
-  );
-  const rightComponent = generateChild(propChildren.rightComponent, 'right');
-
-  childrenArray.push(leftComponent, centerComponent, rightComponent);
-
-  return childrenArray;
-}
+        name={children.icon}
+      />
+    ) : null}
+  </View>
+);
 
-const Header = props => {
-  const {
-    children,
-    statusBarProps,
-    leftComponent,
-    centerComponent,
-    centerComponentStyle,
-    rightComponent,
-    backgroundColor,
-    outerContainerStyles,
-    innerContainerStyles,
-    placement,
-    ...attributes
-  } = props;
-
-  let propChildren = [];
-
-  if (leftComponent || centerComponent || rightComponent) {
-    propChildren = populateChildren(
-      {
-        leftComponent,
-        centerComponent,
-        rightComponent,
-      },
-      placement,
-      centerComponentStyle
-    );
-  }
-
-  return (
-    <View
-      {...attributes}
-      style={[
-        styles.outerContainer,
-        backgroundColor && { backgroundColor },
-        outerContainerStyles,
-      ]}
+const Header = ({
+  statusBarProps,
+  left,
+  center,
+  right,
+  leftContainerStyle,
+  centerContainerStyle,
+  rightContainerStyle,
+  backgroundColor,
+  containerStyle,
+  placement,
+  ...attributes
+}) => (
+  <View
+    {...attributes}
+    style={[
+      styles.container,
+      backgroundColor && { backgroundColor },
+      containerStyle,
+    ]}
+  >
+    <StatusBar barStyle="light-content" {...statusBarProps} />
+    <Children
+      style={[styles.rightLeftContainer, leftContainerStyle]}
+      placement="left"
     >
-      <StatusBar barStyle="light-content" {...statusBarProps} />
-      <View style={[styles.innerContainer, innerContainerStyles]}>
-        {propChildren.length > 0 ? propChildren : children}
-      </View>
-    </View>
-  );
-};
+      {left}
+    </Children>
+    <Children
+      style={[styles.centerContainer, centerContainerStyle]}
+      placement={placement}
+    >
+      {center}
+    </Children>
+    <Children
+      style={[styles.rightLeftContainer, rightContainerStyle]}
+      placement="right"
+    >
+      {right}
+    </Children>
+  </View>
+);
 
 Header.propTypes = {
   placement: PropTypes.oneOf(['left', 'center', 'right']),
```

Comment:
Let's add `placement` to the default props the platform styles, so example on iOS title's are centered and on android titles are left aligned.